### DISCUSSION TITLE
How to download imzML and ibd files programmatically?

### DISCUSSION THREAD

**Discussion by theodev** [metaspace2020/metaspace#1565] [Q&A] [answered]
_This is a historic conversation. A user emailed us in the past:_

Could you please tell me if there is a way to download the .imzML and .ibd files of a given Dataset using the Python package  https://metaspace2020.readthedocs.io/en/latest/ programmatically ?

**Comment by theodev** [comment, marked as answer]
_Our response:_

If you want to download the .imzML and .ibd files of a given dataset using the python package, you can use the function '[download_to_dir](https://metaspace2020.readthedocs.io/en/latest/content/apireference/sm_annotation_utils.html?highlight=download_to_dir#metaspace.sm_annotation_utils.SMDataset.download_to_dir)'. 

Bellow is an example:

```
from metaspace import SMInstance
import getpass

sm = SMInstance(host='https://metaspace2020.eu/')

# if you want to authenticate, uncomment below
# if not sm.logged_in():
#     # Using getpass here prevents the API key from being accidentally saved with this notebook.
#     api_key = getpass.getpass(prompt='API key: ', stream=None)
#     sm.login(api_key=api_key)

sm.dataset(id='ds_id').download_to_dir(path='./')
```

**Reply by yyq83** [reply]
Why does it show “Download_Limit_Reached.txt” when I use this script to download, even though I only downloaded one file?
